### PR TITLE
Adds - CategoricalArrays

### PR DESCRIPTION
**Pull request by aa25desh**
CategoricalArrays & Fix TSF

### CODE REVIEW

File: src/IntervalBasedForest.jl [hunk 1]
```diff
@@ -75,30 +75,24 @@ function InvFeatures(X, n_trees::Int=200, min_interval::Int=3)
...
 
 function proba_predict(forest, X)
-    a = 0
-    for tree in forest
-        if predict(tree, X) == 1
-            a += 1
-        end
-    end
-    (a, length(forest)-a)
+    vv = [predict(tree, X) for tree in forest.tree]
```
Comment (mloning) [inline src/IntervalBasedForest.jl]: I'd prefer to have more meaningful variable names, I have no idea what `vv` stands for?

File: src/IntervalBasedForest.jl [hunk 2]
```diff
@@ -75,30 +75,24 @@ function InvFeatures(X, n_trees::Int=200, min_interval::Int=3)
...
+    #Fix is needed here.
 end
 
 function predict_new(X1, forest)
     t_stamp = size(X1)[1]
     c = Array{Float64}(undef, t_stamp)
```
Comment (mloning) [inline src/IntervalBasedForest.jl]: same here, no sure what `c` is meant to be?

File: src/interface.jl [hunk 3]
```diff
@@ -9,23 +9,25 @@ const IBF = IntervalBasedForest
...
+    integers_seen = MMI.int(classes_seen)
     tree = IBF.randomforestflassifierFit(X, y,
                                     n_trees=m.n_trees,
                                     min_interval=m.min_interval,
                                     pruning_purity_threshold=m.pruning_purity_threshold)
-    fitresult = tree
+    fitresult = (tree, classes_seen, integers_seen)
     cache  = nothing
 
     return fitresult, nothing, nothing
```
Comment (mloning) [inline src/interface.jl:22]: Should that be `return fitresult, cache, report` or so?

**Review by mloning — COMMENTED** [review]
Just a few minor comments!

File: src/interface.jl (hunk 3 above)
Comment (aa25desh) [reply src/interface.jl:22]: Return statement does the same, in our case we don't need `cache, report` yet. & yes `fitresult = (tree, classes_seen, integers_seen)` is correct.

File: src/IntervalBasedForest.jl (hunk 2 above)
Comment (aa25desh) [reply src/IntervalBasedForest.jl]: `c` is the vector of `target` values.

File: src/IntervalBasedForest.jl (hunk 1 above)
Comment (aa25desh) [reply src/IntervalBasedForest.jl]: Yes, thanks for pointing out, will fix naming in next commit:)

**Pull request merged**